src/pgduckdb_detoast.cpp

```diff
@@ -106,15 +111,31 @@ ToastFetchDatum(struct varlena *attr) {
 		SET_VARSIZE(result, attrsize + VARHDRSZ);
 	}
 
-	if (attrsize == 0)
+	if (attrsize == 0) {
 		return result;
+	}
 
 	DuckdbProcessLock::GetLock().lock();
-	toastrel = table_open(toast_pointer.va_toastrelid, AccessShareLock);
-	table_relation_fetch_toast_slice(toastrel, toast_pointer.va_valueid, attrsize, 0, attrsize, result);
-	table_close(toastrel, AccessShareLock);
+	toast_rel = try_table_open(toast_pointer.va_toastrelid, AccessShareLock);
```

Comment:
`try_table_open` might allocate memory an throw an `ERROR` that way. So this needs a `PostgresFunctionGuard` too.